logstash-core/lib/logstash/pipeline_action/reload.rb

```diff
@@ -49,13 +53,35 @@ def execute(agent, pipelines)
 
       logger.info("Reloading pipeline", "pipeline.id" => pipeline_id)
 
-      stop_result = Stop.new(pipeline_id).execute(agent, pipelines)
+      success = pipelines_registry.reload_pipeline(pipeline_id) do
+        # important NOT to explicitly return from block here
+        # the block must return a success boolean value
```

Comment:
I prefer the term "emit" to avoid confusion with the `return` keyword that when used in blocks causes a local-jump with undefined behaviour.

```suggestion
        # the block must emit a success boolean value
```